src/App.jsx

```diff
@@ -16,16 +34,16 @@ function App() {
 
       <div>
         <label htmlFor="userApi">Your API Key:</label>
-        <input type="text" id="userApi" name="userApi" />
+        <input type="text" value={userApiValue} onChange={handleSearchChange} id="userApi" name="userApi" />
       </div>
 
-      <div>
+      {/* <div>
         <label htmlFor="factionFilter">Faction filter:</label>
-        <input type="text" id="factionFilter" name="factionFilter" />
-      </div>
+        <input type="text" value={factionSearchValue} id="factionFilter" name="factionFilter" />
+      </div> */}
 
       <div>
-        <button>Get Data!</button>
+        <button onClick={getData}>Get Data!</button>
```

Comment:
PERF: I recommend adding a `disabled={state === 'loading'}` attribute to this to prevent the user making multiple requests.

Reply:
Added.